### ISSUE TITLE
.well-known/openid-configuration called at each request

### ISSUE THREAD

**Issue by sdementen** [ByteInternet/drf-oidc-auth#32]
hello,

I noticed that the endpoint OIDC_ENDPOINT/.well-known/openid-configuration is called on each request in the `oidc_config` cached_property of `BaseOidcAuthentication`.
This adds consequent latency on each call.
It is used to get the issuer via
```
    @cached_property
    def issuer(self):
        return self.oidc_config['issuer']
```
when validating the jwt at
```
        if id_token.get('iss') != self.issuer:
            msg = _('Invalid Authorization header. Invalid JWT issuer.')
            raise AuthenticationFailed(msg)

```
Could we replace the line above
```
        if id_token.get('iss') != self.issuer:
```
by
```
        if id_token.get('iss') != api_settings.OIDC_ENDPOINT:
```
or would there be unintended consequences of this change ?

**Comment by kerrermanisNL** [comment]
Hi @sdementen I'll have to take a look at it sometime, but I probably won't be a big fan of it. Are you running into performance issues? In general I'd say if it's not really a performance issue why bother? In our case it doesn't really affect speed at all.

**Comment by sdementen** [comment]
Can it at least be cached like the signing keys? But I may be missing something ... (why keys are cached and not the openid-condiguration endpoint)
I also tried to confirm (or not) that the issuer is always equal to the OIDC_ENDPOINT but could not find such confirmation

**Comment by sdementen** [comment]
In terms of performance, it adds the latency of this endpoints to my own endpoint latency which is not acceptable for my use case

**Comment by dedsm** [comment]
I'm revisiting this right now, I made a pull request some time ago to address this: https://github.com/ByteInternet/drf-oidc-auth/pull/27
it works perfectly and I've used it since. now I'm working on making the cache decorator thread-safe

**Comment by kerrermanisNL** [comment]
Hm, must have missed that PR. If you want I can merge it in, or do you want me to wait for the revisited PR?

**Comment by dedsm** [comment]
I think that one works perfectly if you don't care about thread safety.

I'm thinking now on using the django cache framework to do the heavy lifting of handling ttl and by bonus it's thread safe (although with high concurrency it could make some calls to the backend until it's stored in the cache), but that can be a setting, use local or django cache? what do you think?

**Comment by dedsm** [comment]
also there's https://github.com/ByteInternet/drf-oidc-auth/pull/23 that tackles a bit on the threading part

**Comment by dedsm** [comment]
@kerrermanisNL any ETA on the PR(s) ? it would be awesome to have it merged

**Comment by kerrermanisNL** [comment]
Hi @dedsm sorry for the late replies. Unfortunately I'm quite swamped, so I don't have that much time to look at PR's that fall out of my regular working scope. I think I'll have some time on Friday to look at it.

**Comment by kerrermanisNL** [comment]
I think this should be fixed now #27 is merged
